save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/cookie/Cookie.kt

```diff
@@ -2,9 +2,161 @@
 
 package com.saveourtool.save.frontend.externals.cookie
 
+import js.core.jso
 import kotlinext.js.require
 
 /**
  * Object that manages cookies
  */
-val cookie = require("js-cookie")
+val cookie: Cookie = require("js-cookie").unsafeCast<Cookie>()
+
+/**
+ * Interface that encapsulates all cookies interactions
+ */
+external interface Cookie {
+    /**
+     * Get cookie by [key]
+     *
+     * @param key key to get cookie
+     * @param cookieAttribute [CookieAttribute]
+     * @return cookie as [String]
+     */
+    fun get(key: String, cookieAttribute: CookieAttribute = definedExternally): String
+
+    /**
+     * Get all cookies
+     *
+     * @return [Set] of cookies as [String]s
+     */
+    fun get(): Set<String>
+
+    /**
+     * Set cookie
+     *
+     * @param key key to set cookie
+     * @param value cookie value as [String]
+     * @param cookieAttribute [CookieAttribute]
+     */
+    fun set(key: String, value: String, cookieAttribute: CookieAttribute = definedExternally)
+
+    /**
+     * Remove cookie
+     *
+     * @param key cookie key
+     * @param cookieAttribute [CookieAttribute]
+     */
+    fun remove(key: String, cookieAttribute: CookieAttribute = definedExternally)
+}
+
+/**
+ * Cookie attributes that can be passed in [Cookie.remove], [Cookie.set] or [Cookie.get] methods
+ *
+ * @see <a href=https://github.com/js-cookie/js-cookie>Documentation on GitHub</a>
+ */
+external interface CookieAttribute {
+    /**
+     * A [String] indicating the path where the cookie is visible.
+     */
+    var path: String?
+
+    /**
+     * A [String] indicating a valid domain where the cookie should be visible.
+     * The cookie will also be visible to all subdomains.
+     */
+    var domain: String?
+
+    /**
+     * Define when the cookie will be removed.
+     * Value must be an [Int] which will be interpreted as days from time of creation.
+     *
+     * If omitted, the cookie becomes a session cookie.
+     */
+    var expires: Int?
+
+    /**
+     * Either true or false, indicating if the cookie transmission requires a secure protocol (https).
+     */
+    var secure: Boolean?
+}
+
+/**
+ * Class that encapsulates the cookie information
+ *
+ * @property key [String] cookie name
+ * @property expires amount of days before a cookie is considered to be expired, [DEFAULT_EXPIRES] by default
+ */
+sealed class CookieKeys(val key: String, val expires: Int = DEFAULT_EXPIRES) {
+    /**
+     * Cookie that indicates that user as accepted cookie policy
+     */
+    object IsCookieOk : CookieKeys("isCookieOk")
```

Comment:
why sealed classes? Suggest to use Enum and collect all cookies name with description to print these info on cookies page. It allows to avoid changing cookies page when a new cookie is added

Reply:
Our translation is not ready yet. Will do it in next PR

And there probably was a point of why I decided to use `sealed class`, but I do not remember the reason :)

Perhaps it gives us more flexible interface (e.g. create your own cookies during runtime)

Reply:
but for now it seems that there is no actual profit lol

Reply:
Oh sure, I wanted it to have configurable attributes (e.g. `expires` and so on)

So I would leave as is